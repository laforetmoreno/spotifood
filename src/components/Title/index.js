import React from 'react';
import { string } from 'prop-types';

import styles from './index.module.scss';

const Title = ({ title }) => <h1 className={styles.wrapper}>{title}</h1>;

Title.propTypes = {
  title: string,
};

Title.defaultProps = {
  title: '',
};

export default Title;
