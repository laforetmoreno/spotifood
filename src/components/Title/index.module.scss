@import '../../styles';

.wrapper {
  @include flex-default();
  color: $--color-white;
  text-align: center;
  padding: rem(50px);
  margin: 0;
}
