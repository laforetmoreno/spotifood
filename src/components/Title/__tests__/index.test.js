import React from 'react';
import { render, waitForElement, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Title from '../index';

describe('<Title />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Title />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('get by id and change', async () => {
    const { getByTestId } = render(<Title />);
    const fieldNode = await waitForElement(() => getByTestId('title'));
    fireEvent.change(fieldNode);
  });
});
