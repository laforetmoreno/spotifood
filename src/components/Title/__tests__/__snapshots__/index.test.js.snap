// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> renders correctly 1`] = `
<h1
  className="wrapper"
  data-testid="title"
>
  
</h1>
`;
